src/libYARP_OS/src/Timer.cpp

```diff
@@ -0,0 +1,48 @@
+#include <yarp/os/Timer.h>
+#include <yarp/os/Time.h>
+using namespace yarp::os;
+
+Timer::Timer(const double rate, TimerCallback callback, Mutex* mutex, const bool inputOneshot) :
+    RateThread(rate),
+    m_callback(callback),
+    m_oneshot(inputOneshot),
+    m_mutex(mutex)
+{}
+
+
+bool Timer::threadInit()
+{
+    m_startStamp = yarp::os::Time::now();
+    return true;
+}
+
+void Timer::run()
+{
+    if(m_runCount == 0)
+    {
+        m_runCount++;
+        return;
+    }
+
+    YarpTimerEvent event;
+
+    event.currentReal     = yarp::os::Time::now();
+    event.currentExpected = m_startStamp + m_runCount * getRate() * 0.001;
+    event.lastExpected    = event.currentExpected - getRate() * 0.001;
```

Comment:
Not sure about how this will impact performances, but I'd avoid repeating `getRate() * 0.001` twice...